server-v2/api/studio/pkg/auth/authorize.go

```diff
@@ -52,52 +87,53 @@ func AuthMiddlewareWithConfig(config *config.Config) rest.Middleware {
 		return func(w http.ResponseWriter, r *http.Request) {
 			// login handler
 			if strings.HasSuffix(r.URL.Path, "/connect") {
-				fmt.Println("=====global middleware", r.URL.Path)
 				var req types.ConnectDBParams
-				err := httpx.Parse(r, &req)
+				rClone := utils.CopyHttpRequest(r)
+				err := httpx.Parse(rClone, &req)
 				if err != nil {
-					fmt.Println("=====req3333", req)
+					http.Error(w, err.Error(), http.StatusBadRequest)
+					return
 				}
-				fmt.Println("=====err", err)
-				fmt.Println("=====req.Address", req.Address)
-				fmt.Println("=====req.Port", req.Port)
-				fmt.Println("=====req.Authorization", req.Authorization)
-			}
-			c1 := http.Cookie{
-				Name:     "access_token",
-				Value:    "12333",
-				Path:     "/",
-				HttpOnly: true,
-				MaxAge:   3600,
-			}
 
-			var req1 types.ConnectDBParams
-			err1 := httpx.Parse(r, &req1)
-			fmt.Println("=====err1=====", err1)
+				tokenString, clientInfo, err := parseConnectDBParams(&req, config)
+				if err != nil {
+					http.Error(w, err.Error(), http.StatusBadRequest)
+					return
+				}
 
-			w.Header().Set("Access-Control-Allow-Origin", "*")
-			w.Header().Set("Set-Cookie", c1.String())
-			next(w, r)
-		}
-	}
-}
+				token := http.Cookie{
+					Name:     "token",
+					Value:    tokenString,
+					Path:     "/",
+					HttpOnly: true,
+					MaxAge:   1800,
+				}
+				nsid := http.Cookie{
+					Name:     "nsid",
```

Comment:
In order to make the nsid conflict with other apps, you'd better make it configuarable and add a prefix like `explorer-nsid`. So same is the cookie field `token`